interceptors/elevated_privileges.go

```diff
@@ -0,0 +1,34 @@
+package interceptors
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/companieshouse/chs.go/log"
+	"github.com/companieshouse/payments.api.ch.gov.uk/helpers"
+)
+
+// ElevatedPrivilegesInterceptor checks that the user is authenticated
+func ElevatedPrivilegesInterceptor(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Check headers for identity type and identity
+		identityType := helpers.GetAuthorisedIdentityType(r)
+		if !(identityType == helpers.APIKeyIdentityType) {
+			log.Error(fmt.Errorf("elevated privileges interceptor unauthorised: not API key identity type"))
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		isAPIKeyRequest := identityType == helpers.APIKeyIdentityType
```

Comment:
this has already been checked above so just need to check for elevated privileges here